packages/go_router/lib/src/parser.dart

```diff
@@ -158,7 +161,13 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
         });
       },
       onCanNotEnter: () {
-        // If blocked, "stay" on last successful match if available.
+        // If blocked, stay on the current route by restoring the last known good configuration.
```

Comment:
This new check  looks good to me! 

also I think it's ok to remove the old check in line 171 for `_lastMatchList` if we are checking "router.routerDelegate.currentConfiguration" already?